### DISCUSSION TITLE
Lesson 2: Confusion regarding array declaration

### DISCUSSION THREAD

**Discussion by AyuK03** [smartcontractkit/full-blockchain-solidity-course-py#1732] [Q&A] [answered]
In the code snippet : **SimpleStorage[] public simpleStorageArray;**

I am unsure as to what the data type of the array is. The term "SimpleStorage" before [] brackets depict the entire contract that was made in the lesson 1 right ? Can a contract be passed as a data type for an array ? 

Kindly help me clear this doubt. I am confused as to why SimpleStorage is an array's data type.

**Comment by cromewar** [comment, marked as answer]
Hello @AyuK03 
You are right, in other programming languages the contract is like a Class, so you can create a group of it into an array.
So, you have SimpleStorage contract, and if you create `SimpleStorage[] public simpleStorageArray` you are creating a group of Simple Storage contracts, nice isn't it?

**Reply by AyuK03** [reply]
Yes now I get a clearer view. We are using Simple storage as a class and thus creating multiple contract objects that are getting stored in the array. 

Thank you @cromewar 🙂
